specification/dartLangSpec.tex

```diff
@@ -8848,53 +8851,84 @@ \subsubsection{Unqualified Invocation}
 }
 
 \LMHash{}%
-It is a compile-time error if $i$ occurs inside a top-level or static function
-(be it function, method, getter, or setter)
-or a top-level or static variable initializer,
-and there is no lexically visible declaration named \id{} in scope.
+For the static analysis of $i$,
+a lexical lookup of \id{} is performed
+(\ref{lexicalLookup}):
 
 \LMHash{}%
-If there exists a lexically visible declaration named \id,
-let $D_{id}$ be the innermost such declaration.
-Then:
+\Case{Lexical lookup yields a declaration}
+In this case the lexical lookup of \id{}
+yields a declaration $D$.
 \begin{itemize}
-\item Consider the situation where $D_{id}$ is a type declaration.
-  If $D_{id}$ is a declaration of a class $C$
+\item
+  Consider the case where $D$ is a type declaration.
+  If $D$ is a declaration of a class $C$
   that has a constructor named $C$
   then the meaning of $i$ depends on the context:
   If $i$ occurs in a constant context
   (\ref{constantContexts}),
-  then $i$ is equivalent to \code{\CONST\,\,$i$};
+  then $i$ is treated as
+  (\ref{notation})
+  \code{\CONST\,\,$i$};
   if $i$ does not occur in a constant context
-  then $i$ is equivalent to \code{\NEW\,\,$i$}.
+  then $i$ is treated as \code{\NEW\,\,$i$}.
   Otherwise a compile-time error occurs
-  (\commentary{that is, if $D_{id}$ does not declare a class,
+  (\commentary{that is, if $D$ does not declare a class,
     or it declares a class that has no constructor named $C$}).
-\item Otherwise, if $D_{id}$ is an import directive
-  where \id{} is declared to be a library prefix,
-  a compile-time error occurs.
-\item Otherwise, if $D_{id}$ declares
+\item
+  Otherwise, if $D$ is a declaration of
   a local function,
   a library function, or
   a library or static getter, or a variable,
-  then $i$ is treated as a function expression invocation
+  then $i$ is treated as
+  (\ref{notation})
+  a function expression invocation
   (\ref{functionExpressionInvocation}).
-\item Otherwise, if $D_{id}$ is a static method of the enclosing class $C$,
-  $i$ is equivalent to
-  \code{$C$.\id<$A_1, \ldots,\ A_r$>($a_1, \ldots,\ a_n,\ x_{n+1}$: $a_{n+1}, \ldots,\ x_{n+k}$: $a_{n+k}$)}.
-\item Otherwise, if $i$ occurs in an instance method body,
-  $i$ is equivalent to the ordinary method invocation
-
-  \code{\THIS{}.\id<$A_1, \ldots,\ A_r$>($a_1, \ldots,\ a_n,\ x_{n+1}$: $a_{n+1}, \ldots,\ x_{n+k}$: $a_{n+k}$)}.
+\item
+  Otherwise, if $D$ is a static method, getter, or variable declaration
+  in the enclosing class $C$,
```

Comment:
enclosing class -> enclosing class or mixin

(and soon extension)

Reply:
Indeed! Done.